### ISSUE TITLE
Bug: function get_table_rows_by_seckey call wrong function convert_to_type

### ISSUE THREAD

**Issue by gzliudan** [EOSIO/eos#9195]
https://github.com/EOSIO/eos/blob/master/plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp#L452
SecKeyType lv = convert_to_type<SecKeyType>( s.to_string(), "lower_bound name" ); 

The above function convert_to_type which being called is https://github.com/EOSIO/eos/blob/master/plugins/chain_plugin/chain_plugin.cpp#L1648
It first convert string to number, if fails then convert string to name and convert name to unit64_t.

This make function get_table_rows_by_seckey works abnormally(return null result) when lowerbound or upperbound string is all numbers. In fact string should be convert to name first, and get uint64_t value.

**Comment by allenhan2** [comment]
Proposed solution is adding a trailing "_n" to differentiate digits from all-digits name. It is not ideal as areg pointed out, and will work as a temporary solution before cleos implements a mechanism to enforce the parameter types before reaching backend. 
The solution should have no impact on existing code, since digits still can be converted properly, but provides a leeway to allow all-digits name can be processed properly as a name.

**Comment by allenhan2** [comment]
@gzliudan  could you please provide a user scenario for this issue also? thanks a lot!

**Comment by allenhan2** [comment]
Look for alternative of using DB index type for conversion now instead of appending trailing "_n"

**Comment by gzliudan** [comment]
> @gzliudan could you please provide a user scenario for this issue also? thanks a lot!

for example: there is a multi_index table which name is mytable, the define is
```cpp
struct mytable {
    uint64_t pkey;
    name account;
    int  age;
    uint64_t primary_key() const { return pkey; }
    uint64_t get_account() const { return account.value; }
};
```
And this table has a record: pkey=(any value), account=“12345”_n，when I want to get all fields of this record by account field, I run below command:

`cleos get table {MY-CODE} {MY-SCOPE} mytable --index 2 --key-type name -L 12345 -U 12345`
or
`curl http://127.0.0.1:8888/v1/chain/get_table_rows -X POST -d '{"json":true,"code":"MY-CODE","scope":"MY-SCOPE","table":"mytable","lower_bound":"12345","upper_bound":"12345","index_position":"sec","key_type":"name","limit":10,"reverse":false,"show_payer":false}'`

then command returns no data, but mytable has a record  in fact .

**Comment by gzliudan** [comment]
only eosio.cdt support "12345"_n; eos does not support "12345"_n, only support N(12345) .  
I suggest create a new function use other name instead of  convert_to_type. Below version should be right:
https://github.com/EOSIO/eos/blob/master/plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp#L58

> Proposed solution is adding a trailing "_n" to differentiate all-digits name from digits.
> The solution should have no impact on existing code, since digits still can be converted properly, but provides a leeway to allow all-digits name can be processed properly as a name.
> Example:
> "1234" --> integer 1234
> "1234_n" --> name "1234"
> 
> Note:
> This rule only applies to all digits with following '_n".
> For example:
> "abc_n", "a12_n", "12.34_n", trailing "_n" won't be removed, and will cause conversion failing

**Comment by allenhan2** [comment]
@gzliudan appreciate for the detailed explanation,

**Comment by allenhan2** [comment]
@gzliudan 
I have not work on this issue due to another higher priority one. 
Modified get_table_tests.cpp, to query all digits "newname", seems working as expected when setting lower_bound and upper_bound as all digits. 
How to replicate:
1. branch : develop
2. add this line " bidname(N(inita), N(111), eosio::chain::asset::from_string("1.0000 SYS"));"  at get_table_tests.cpp:363
3. add those two lines at get_table_tests.cpp:373
   p.lower_bound = "111";
   p.upper_bound = "111";
   p.index_position = "primary";
   p.key_type = "name";
4. recompile, you will see only one record being fetched.

Maybe, I missed something after not working on it for a while. Let me know what you think

**Comment by gzliudan** [comment]
I believe it‘s a small and low priority bug. It's not important, because I can change key-type not use name to avoid riggering this problem. You can invest this issue at any time later.

BTW: where can I find this year's road map of EOS?

**Comment by taokayan** [comment]
a workaround is to add a space before the string, for example " 1111" instead of "1111"

**Comment by gzliudan** [comment]
How about eos v2.1.0 ?

**Comment by sanaraufx** [comment]
Fixed in 2.1 release
